src/axom/primal/tests/primal_polygon.cpp

```diff
@@ -3,11 +3,13 @@
 //
 // SPDX-License-Identifier: (BSD-3-Clause)
 
-#include "gtest/gtest.h"
-
+#include "axom/config.hpp"
```

Comment:
I suspect the "Undefined M_PI" error in MSVC was due to including `gtest/gtest.h` before `axom/config.hpp`, which defined `_USE_MATH_DEFINES`.

There's an extra include guard in `<cmath>` that requires `_USE_MATH_DEFINES` to be set before the first time `<cmath>` is included.

See, e.g.: 
* https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio
* https://github.com/search?q=repo%3Agoogle%2Fgoogletest%20cmath&type=code

Reply:
Errors were of the form:
```
2023-06-19T18:33:43.8255199Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(603,31): error C2065: 'M_PI': undeclared identifier [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
2023-06-19T18:33:43.8255668Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(603,18): error C2737: 'alpha': const object must be initialized [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
2023-06-19T18:33:43.8256083Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(612,39): error C2065: 'M_PI': undeclared identifier [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
2023-06-19T18:33:43.8256465Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(613,39): error C2065: 'M_PI': undeclared identifier [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
2023-06-19T18:33:43.8256870Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(614,43): error C2065: 'M_PI': undeclared identifier [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
2023-06-19T18:33:43.8257267Z   D:\a\1\s\src\axom\primal\tests\primal_polyhedron.cpp(615,43): error C2065: 'M_PI': undeclared identifier [D:\a\1\s\build\axom\primal\tests\primal_polyhedron_test.vcxproj]
```
https://dev.azure.com/healy22/d4445b8f-6e95-49cd-9e5b-8c8ca1ccad31/_apis/build/builds/8213/logs/39